integration-tests/smoke/ccip/ccip_add_chain_test.go

```diff
@@ -0,0 +1,851 @@
+func setupChain(t *testing.T, e ccipcs.DeployedEnv, tEnv ccipcs.TestEnvironment, chains []uint64, deployJobs, deployHomeChain, mcmsEnabled bool) ccipcs.DeployedEnv {
+	e = ccipcs.AddCCIPContractsToEnvironment(
+		t,
+		chains,
+		tEnv,
+		deployJobs,
+		deployHomeChain,
+		mcmsEnabled,
+	)
+
+	// Need to update what the RMNProxy is pointing to, otherwise plugin will not work.
+	var err error
+	e.Env, err = commonchangeset.ApplyChangesets(t, e.Env, e.TimelockContracts(t), []commonchangeset.ChangesetApplication{
+		{
+			Changeset: commonchangeset.WrapChangeSet(ccipcs.SetRMNRemoteOnRMNProxy),
+			Config: ccipcs.SetRMNRemoteOnRMNProxyConfig{
+				ChainSelectors: chains,
+			},
+		},
+	})
+	require.NoError(t, err)
+
+	return e
+}
+
+// assertChainWiringInbound checks that the new chain has the existing chains enabled as sources.
+// It only checks the inbound wiring on the new chain.
+// It doesn't check that the existing chains have the new chain enabled as a dest.
+func assertChainWiringInbound(
+	t *testing.T,
+	state ccipcs.CCIPOnChainState,
+	newChain uint64,
+	existingChains []uint64,
+	testRouterEnabled bool,
+) {
+	for _, existingChain := range existingChains {
+		var rtr *router.Router
+		if testRouterEnabled {
+			rtr = state.Chains[newChain].TestRouter
+		} else {
+			rtr = state.Chains[newChain].Router
+		}
+
+		// check that the offRamp has the existing chain enabled as a source.
+		dcc, err := state.Chains[newChain].OffRamp.GetSourceChainConfig(&bind.CallOpts{
+			Context: tests.Context(t),
+		}, existingChain)
+		require.NoError(t, err)
+		require.Equal(t, rtr.Address(), dcc.Router)
+
+		// check that the router has the existing chain enabled as a source.
+		routerOffRamps, err := rtr.GetOffRamps(&bind.CallOpts{
+			Context: tests.Context(t),
+		})
+		require.NoError(t, err)
+
+		var found bool
+		for _, offRamp := range routerOffRamps {
+			if offRamp.SourceChainSelector == existingChain {
+				require.Equal(t, state.Chains[newChain].OffRamp.Address(), offRamp.OffRamp)
+				found = true
+				break
+			}
+		}
+		require.True(t, found)
+	}
+}
+
+// assertChainWiringOutbound checks that the new chain can be requested from the existing chains.
+// This only checks the outbound wiring on the existing chains.
+// It doesn't check that the new chain can process the requests.
+func assertChainWiringOutbound(
+	t *testing.T,
+	state ccipcs.CCIPOnChainState,
+	newChain uint64,
+	existingChains []uint64,
+	testRouterEnabled bool,
+) {
+	for _, existingChain := range existingChains {
+		var rtr *router.Router
+		if testRouterEnabled {
+			rtr = state.Chains[existingChain].TestRouter
+		} else {
+			rtr = state.Chains[existingChain].Router
+		}
+
+		// check that the onRamp has the new chain enabled as a dest.
+		dcc, err := state.Chains[existingChain].OnRamp.GetDestChainConfig(&bind.CallOpts{
+			Context: tests.Context(t),
+		}, newChain)
+		require.NoError(t, err)
+		require.Equal(t, rtr.Address(), dcc.Router)
+
```

Comment:
should this be the other way? should it not check if offRamp of existing chains have wiring for newchain?

Reply:
This is checking that users on `existingChains` can send messages to `newChain`, so the check would be the onramps, fee quoters, and router (i.e messages going _out_ of existing chains _to_ new chain). Checking the offramp would be the other way around - checking that `existingChains` can receive messages from `newChain`.